### PR TITLE
Break out chat and threads from nip 29

### PR DESCRIPTION
**Pull request by staab**
This is a variant of https://github.com/nostr-protocol/nips/pull/1589 which attempts to reconcile NIP 29 and flotilla's rooms over chat and thread kinds. This requires no changes of NIP 29 clients, only flotilla will need to be modified.

I originally chose new event kinds in order to avoid stepping on anyone's toes, but it would be good if we can reconcile the two. One thing that is important to me for the semantics of chat/threads is that arbitrary reply hierarchies need not be supported. If NIP 29 folks are ok with the "shallow reply hierarchies" stipulation, then we can go ahead with this.

### CODE REVIEW

**Comment by staab** [comment]
Just updated this to only move kind 9 and 11 out of NIP 29.

File: 7D.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+NIP-&D
```
Comment (fiatjaf) [inline 7D.md:1]:
```suggestion
NIP-7D
```

File: 7D.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+A thread is a `kind 11` event.  Threads SHOULD include a `title` with a summary
+of the thread's topic.
+
+```json
+{
+  "kind": 11,
+  "content": "Good morning",
+  "tags": [
+    ["title", "GM"]
```
Comment (fiatjaf) [inline 7D.md:17]:
```suggestion
    ["subject", "GM"]
```

Maybe?

**Review by fiatjaf — APPROVED** [review]

**Comment by fiatjaf** [comment]
Please remember to squash instead of merging.

**Comment by laanwj** [comment]
> Are NIPs hex now? holy fuck

Hex, or base36,  or who knows?

FWIW, adding letters breaks the auto-link regexp for NIP-XX for at least nostrudel, not a big deal but it'd be kind of useful to know what to expect, so we can be prepared when say BIP-🐙🦑 is added.

**Pull request merged**